centrallix/objectsystem/obj_datatypes.c

```diff
@@ -1470,30 +1479,21 @@ objDataToMoney(int data_type, void* data_ptr, pMoneyType m)
 		    }
 		if (*endptr == '-')
 		    {
-		    m->WholePart = -m->WholePart;
+		    m->Value = -m->Value;
 		    is_neg = !is_neg;
 		    }
-		if (is_neg && m->FractionPart != 0)
-		    {
-		    m->WholePart--;
-		    m->FractionPart = 10000 - m->FractionPart;
-		    }
 		break;
 	    
 	    case DATA_T_DOUBLE:
-	        dbl = *(double*)data_ptr + 0.000001;
-	        m->WholePart = floor(dbl);
-		m->FractionPart = (dbl - m->WholePart)*10000;
+            m->Value = *(double*)data_ptr * 10000.0;
```

Comment:
A couple things:
- Do you know why the original implementation adds 0.000001 to the double? Is that something you should still do?
- Are you sure that m->Value is going to end up as a proper long long type this way, or does it need to be multiplied by 10000 (without the decimal place) or cast?

Reply:
This works, but you need to add 0.1 (if positive, or subtract 0.1 if negative) to the value **before** assignment to the long long in the event that `*(double*)data_ptr * 10000.0` ends up something like 9999.999999999999 instead of 10000.0 for example.  Floating point arithmetic is imprecise in a way that can cause this sort of round-off error, and conversion back to an integer value needs a fudge factor included to prevent unexpected truncation.